runtime/compiler/control/MongoLogger.cpp

```diff
@@ -0,0 +1,161 @@
+#include "MongoLogger.hpp"
+#include <ctime>
+#include <stdio.h>
+#include <string.h>
+#include <string>
+#include <stdlib.h>
+#include "LoadDBLibs.hpp"
+
+MongoLogger::MongoLogger(const char* databaseIP, uint32_t databasePort, const char* databaseName)
+      : BasePersistentLogger(databaseIP, databasePort, databaseName)
+   {
+   init();
+   }
+
+MongoLogger::MongoLogger(const char* databaseIP, uint32_t databasePort, const char* databaseName,
+                         const char* databaseUsername, const char* databasePassword)
+      : BasePersistentLogger(databaseIP, databasePort, databaseName, databaseUsername, databasePassword)
+   {
+   init();
+   }
+
+MongoLogger::~MongoLogger()
+   {
+   //Clean up this mongoc logger.
+   Omongoc_collection_destroy(_collection);
+   Omongoc_database_destroy(_db);
+   Omongoc_client_destroy(_client);
+   Omongoc_uri_destroy(_uri);
+   }
+
+void MongoLogger::init()
+   {
+   _uri = NULL;
+   _collection = NULL;
+   _db = NULL;
+   _client = NULL;
+   }
+
+char * MongoLogger::constructURI()
+   {
+   // Check if we have the database name
+   if (strcmp(_databaseName,"") == 0)
+      {
+      return "jitserver_logs";
```

Comment:
In most cases we write something into `_uri_string`, but in this case we don't. For uniformity I would copy the database name into the `_uri_string` buffer.